src/core/hooks/use-rails-ujs-hooks.ts

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useRef, RefObject } from "react";
+
+type HttpMethod = "get" | "post" | "put" | "patch" | "delete";
+
+interface RailsUjsLink extends HTMLAnchorElement {
+  dataset: {
+    method?: string;
+    confirm?: string;
+    remote?: string;
+  };
+}
+
+interface CsrfMetaTag extends HTMLMetaElement {
+  content: string;
+}
+
+const useRailsUjsLinks = (): RefObject<HTMLDivElement> => {
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const container = containerRef.current;
+    if (!container) return;
```

Comment:
How does this work? The ref is set on a span elsewhere (not a div, so the type on L18 isn't right), but is already formed and this logic only runs once on mount. `containerRef` instantiates with `{ current: null }` so this shouldn't pass - unless I have a misunderstanding about how this works or effects elsewhere.

The way you usually see it is to make your ref, add it to an element, then have a useEffect dependent on ref.current that does stuff when the two have reconciled and mounted.

This must re-run, but I'm not sure how.

Reply:
AI slop and I just let it in, I've added a fixup where I now pass in the ref rather than creating it